ql/src/java/org/apache/hadoop/hive/ql/session/SessionState.java

```diff
@@ -349,6 +349,20 @@
 
   private final AtomicLong sparkSessionId = new AtomicLong();
 
+  private final Map<String, Object> queryStateMap = new HashMap<>();
+
+  public Object getQueryState(String queryId) {
+    return queryStateMap.get(queryId);
+  }
+
+  public void addQueryState(String queryId, Object queryState) {
```

Comment:
Shouldn't we only accept only QueryState objects here?
i.e. `public void addQueryState(String queryId, QueryState queryState) {`

Reply:
Right. Changed it.